caddyconfig/httpcaddyfile/options_test.go

```diff
@@ -0,0 +1,114 @@
+package httpcaddyfile
+
+import (
+	"testing"
+
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+	_ "github.com/caddyserver/caddy/v2/modules/logging"
+)
+
+func TestGlobalLogOptionSyntax(t *testing.T) {
```

Comment:
This is testing the global options in isolation, which is good, but I think we should also test global logs in combination with logs configured in sites.

An easy way to do that is by writing an adapt test, you can find the ones we have already here: https://github.com/caddyserver/caddy/tree/master/caddytest/integration/caddyfile_adapt

You could maybe move all of the tests in this file to be adapt test files instead, they're a bit easier to read and update, and render a diff of the JSON on error.

Reply:
Great, thanks for pointing that out. I ended up splitting the tests so that the more complex successful test cases are in the adapt folder, and the failure cases remain unit tests, since it seems like the adapt folder does not support that? Let me know if this matches what you'd expect.

Reply:
Yeah that's perfect, thanks

Reply:
I think we're still missing an adapt test that has both a global logger and a logger in a site. Something like this:

```
{
	log {
		output file caddy.log
	}
}

:8884 {
	log {
		output file access.log
	}
}
```

Reply:
Great idea, and it turns out that caught a subtle edge-case bug the default logger would end up with logs configured for it excluded by this logic: https://github.com/caddyserver/caddy/blob/51f35ba03f78e691d0e141b24ff19f82bfb34b67/caddyconfig/httpcaddyfile/httptype.go#L313-L322

That case is now handled better and a new test case file covers it.

Reply:
Perfect, thank you!